cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java

```diff
@@ -87,6 +98,19 @@ public void validate(AmazonIdentityManagementClient iam, CloudS3View cloudFileSy
             SortedSet<String> failedActions = new TreeSet<>();
             SortedSet<String> warnings = new TreeSet<>();
             List<Policy> policies = collectPolicies(cloudFileSystem, policyFileNames);
+            if ((entitlementService.isDatalakeBackupRestorePrechecksEnabled(ThreadBasedUserCrnProvider.getAccountId())) &&
```

Comment:
this entire block should be in the collectPolicies function, or better in a different function called by collectPolicies